content/blog/_posts/2021-05-java-client.md

```diff
@@ -0,0 +1,660 @@
+---
+title: Wrapping your rust in a Java Jar for distribution
+author:
+    name: "Sebastian Imlay"
+    github: "simlay"
+description: "Learn how to wrap your Rust crate in a Java Jar for the desktop/server"
+date: 2021-05-03
+slug: java-client
+url: /blog/2021/05/java-client
+img: blog/images/java-client/crab-in-jar.png
+img-credit:
+    link: https://www.flickr.com/photos/jenny-pics/4821968234/
+    author: Jenny Downing
+    site: Flikr
+twitter-card: summary_large_image
+code:
+    height: 1200
+---
+
+This week, we're happy to announce the addition of a [Java client library for Fluvio].
+Using the Java client is just as easy as using our other clients. Check out the
+[hello world in Java tutorial] or [documentation] for usage.
+
+This post will talk about how we bundled and distributed our rust code into a [Java Jar] using [Gradle] to target a desktop enviroment.  To do
+this for android, we recommend the [Flapigen android example].
+
+[Java client library for Fluvio]: https://github.com/infinyon/fluvio-client-java
+[hello world in Java tutorial]: /tutorials/java/hello-world/
+[documentation]: https://infinyon.github.io/fluvio-client-java/com/infinyon/fluvio/package-summary.html
+[Java Jar]: https://docs.oracle.com/javase/tutorial/deployment/jar/basicsindex.html
+[Gradle]: https://gradle.org/
+[Flapigen android example]: https://dushistov.github.io/flapigen-rs/java-android-example.html
+
+# Overview
+
+Similar to [our python post], we used [flapigen] for packaging, but then it took a bit of research and many cups of coffee to figure out how to bundle rust into a jar for publishing. Let's get started.
```

Comment:
I think it may be helpful to add a few bullet points about the handful of steps you're going to be covering, that way readers sort of have a roadmap while reading. Something like:

- Setting up Gradle and Rust projects
- Configuring flapigen to run at build-time and generate Java bindings from our Rust interface
- Bundling the Rust native library into the Java jar
- Loading the native library at runtime
- Writing tests to verify the behavior

Reply:
I added basically these exact words with links to their sections.